path/__init__.py

```diff
@@ -1164,6 +1135,28 @@ def chmod(self, mode):
         os.chmod(self, mode)
         return self
 
+    def lchmod(self, mode):
+        """
+        Set the mode. May be the new mode (os.chmod behavior) or a `symbolic
+        mode <http://en.wikipedia.org/wiki/Chmod#Symbolic_modes>`_.
+
+        .. seealso:: :func:`os.chmod`
+        """
+        if isinstance(mode, string_types):
+            mask = _multi_permission_mask(mode)
+            mode = mask(self.stat().st_mode)
+        os.lchmod(self, mode)
+        return self
+
+    if 'grp' in globals():
+        def group(self):
+            if self.exists():
```

Comment:
Nitpick: I'd write the exceptional condition as a short-circuit and then the main logic in the main level of the code. i.e.:

```
def group(self):
    if not self.exists():
        return
    gid = os.stat(self)
    ...
    return group_data.gr_name
```